db/models.go

```diff
@@ -16,14 +16,15 @@ type UserConfig struct {
 }
 
 type App struct {
-	ID          uint
-	Name        string `validate:"required"`
-	Description string
-	NostrPubkey string `validate:"required"`
-	CreatedAt   time.Time
-	UpdatedAt   time.Time
-	Isolated    bool
-	Metadata    datatypes.JSON
+	ID             uint
+	Name           string `validate:"required"`
+	Description    string
+	NostrPubkey    string `validate:"required"`
+	walletChildIdx uint   `validate:"required"`
+	CreatedAt      time.Time
+	UpdatedAt      time.Time
+	Isolated       bool
+	Metadata       datatypes.JSON
```

Comment:
How do we ensure the old app connections work using the original wallet service key? do we need an app version field?

Reply:
We should be able to compare the p tag on the request events to keys.GetNostrPublicKey, if it matches it means it is an old app connection still not using child keys, right?

Reply:
Hmm, but when starting up the hub we actually don't know if a certain app from DB is an old version or new with child key, and thus can't decide if we need to open a separate subscription for that app or not.
So it might be actually needed some kind of flag/version on the app model.

Reply:
Maybe we add a field with wallet child pubKey which we only fill for new app connections created with child derived keys?
We would avoid then having to calculate each child derived key from master priv key (and from that the corresponding nostr pubKey) during the starting phase of the hub when we only need those apps pubKeys for subscribing to nwc requests on the relay.
Wdyt?

Reply:
@frnandu that sounds good to me. We then only need to derive the child private keys when creating events, right?

Reply:
I think also when handling events for `nip04.ComputeSharedSecret(event.PubKey, appWalletPrivKey)`

Reply:
done